sdk/eventhub/Azure.Messaging.EventHubs/src/EventData.cs

```diff
@@ -28,6 +29,29 @@ public class EventData
         ///
         public ReadOnlyMemory<byte> Body { get; }
 
+        /// <summary>
+        ///   The data associated with the event, in stream form.
+        /// </summary>
+        ///
+        /// <value>
+        ///   A <see cref="Stream" /> containing the raw data representing the <see cref="Body" />
+        ///   of the event.  The caller is assumed to have ownership of the stream, including responsibility
+        ///   for managing its lifespan and ensuring proper disposal.
+        /// </value>
+        ///
+        /// <remarks>
+        ///   If the means for deserializaing the raw data is not apparent to consumers, a
+        ///   common technique is to make use of <see cref="EventData.Properties" /> to associate serialization hints
+        ///   as an aid to consumers who wish to deserialize the binary data.
+        /// </remarks>
+        ///
+        /// <seealso cref="EventData.Properties" />
+        ///
+        public Stream BodyAsStream
```

Comment:
Generally, you want to drop to the lowest common type rather than being as specific as possible, unless there's a good reason for that specificity.   The underlying justification is that you want to give your implementation as much freedom to change as possible without introducing unnecessary changes to the public API.

The design principle comes down to asking yourself "What is the actual contract that we want to guarantee for callers?"   In this case (and others in the SDK) we want to allow streaming semantics for library users but don't want to offer the guarantee that the stream is memory-based.  In the future, we may want to stream that in from the network or some intermediary buffer.

For something concrete, in the case of Event Hubs, think about introducing a pipeline that events pass through before being returned which may do things like encrypt/decrypt message bodies.   Those aren't necessarily memory streams.  We would either have to decide that the specific `MemoryStream` interface was important to our users and force each stream from the pipeline into it, potentially allocating and forcing local buffering that could eat resources, or we could keep the loose guarantee of "this thing is some kind of stream.  Ask it if you need capabilities like seeking to see if they're supported."